packages/devextreme/testing/tests/DevExpress.ui.widgets/multiView.tests.js

```diff
@@ -1659,3 +1659,127 @@ QUnit.module('swipeable disabled state', () => {
+            selectedIndex: 2
+        });
+        const instance = $multiView.dxMultiView('instance');
+        instance.option({
+            items: [
+                { text: '1', visible: false },
+                { text: '2', visible: false },
+                { text: '3', visible: false },
+            ]
+        });
+
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is updated on proper index');
+    });
+
+    QUnit.test('when hiding non-selected item selectedIndex should change to first visible item', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: true },
+                { text: '3', visible: true },
+            ],
+            selectedIndex: 2
+        });
+        const instance = $multiView.dxMultiView('instance');
+        instance.option('items[1].visible', false);
+
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is updated on proper index');
+    });
+
+    QUnit.test('when hiding selected item selectedIndex should change to first visible item', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: true },
+                { text: '3', visible: true },
+            ],
+            selectedIndex: 1
+        });
+        const instance = $multiView.dxMultiView('instance');
+        instance.option('items[1].visible', false);
+
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is updated on proper index');
+    });
+
+    QUnit.test('when showing previous invisible items selectedIndex should change to first visible item', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: false },
+                { text: '3', visible: true },
+            ],
+            selectedIndex: 2
+        });
+        const instance = $multiView.dxMultiView('instance');
+        instance.option('items[1].visible', true);
+
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is updated on proper index');
+    });
+
+    QUnit.test('when hiding last visible item selectedIndex should return to 0 index', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: false },
+                { text: '2', visible: false },
+                { text: '3', visible: true },
+            ],
+            selectedIndex: 2
+        });
+        const instance = $multiView.dxMultiView('instance');
+        instance.option('items[2].visible', false);
+
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is updated on proper index');
+    });
+});
```

Comment:
[testing coverage]
- next visible item should be selected if currently selected item is hidden and loop=true | items=[{ visible: true }, { visible: false }], selectedIndex = 1
- when hiding non-selected item after current selectedIndex, selectedIndex should not be changed
- when showing previously invisible item after selectedIndex, selectedIndex should not be changed
- when hiding selected item positioned in the end, next visible item to the left is selected

Reply:
[08ac2cc](https://github.com/DevExpress/DevExtreme/pull/28145/commits/08ac2ccbd0c1b796250708330fe3c53877470a0c)

Reply:
[89c8df9](https://github.com/DevExpress/DevExtreme/pull/28145/commits/89c8df9852c38a344cd8ae4768d596c93182fab2)